test/berlim_web/controllers/admin_controller_test.exs

```diff
@@ -1,87 +1,78 @@
 defmodule BerlimWeb.AdminControllerTest do
   use BerlimWeb.ConnCase
-  use BerlimWeb.Helpers.AuthHelper
 
   import Berlim.Factory
+  alias Berlim.InternalAccounts.Admin
 
   @create_attrs params_for(:admin)
   @update_attrs %{name: "Lionel Ritchie"}
   @invalid_attrs %{name: nil, email: nil}
 
-  describe "GET /index, when user is an admin" do
-    setup [:authenticate_admin]
+  setup %{conn: conn} do
+    {:ok, conn: put_req_header(conn, "accept", "application/json")}
+  end
 
+  describe "GET /index" do
     test "lists all admins", %{conn: conn} do
       conn = get(conn, Routes.admin_path(conn, :index))
-      assert html_response(conn, 200) =~ "Admins"
+      assert json_response(conn, 200)["data"] == []
     end
   end
 
-  describe "GET /new, when user is an admin" do
-    setup [:authenticate_admin]
-
-    test "renders form", %{conn: conn} do
-      conn = get(conn, Routes.admin_path(conn, :new))
-      assert html_response(conn, 200) =~ "Novo Admin"
-    end
-  end
+  describe "POST /create" do
+    test "renders admin show when data is valid", %{conn: conn} do
+      conn = post(conn, Routes.admin_path(conn, :create), admin: @create_attrs)
+      assert %{"id" => id} = json_response(conn, 201)["data"]
 
-  describe "POST /create, when user is an admin" do
-    setup [:authenticate_admin]
+      conn = get(conn, Routes.admin_path(conn, :show, id))
 
-    test "redirects to index when data is valid", %{conn: conn} do
-      conn = post(conn, Routes.admin_path(conn, :create), admin: @create_attrs)
-      assert redirected_to(conn) == Routes.admin_path(conn, :index)
+      assert %{
+               "id" => id,
+               "name" => "John Doe"
+             } = json_response(conn, 200)["data"]
     end
 
     test "renders errors when data is invalid", %{conn: conn} do
       conn = post(conn, Routes.admin_path(conn, :create), admin: @invalid_attrs)
-      assert html_response(conn, 200) =~ "Novo Admin"
-    end
-  end
-
-  describe "GET /edit, when user is an admin" do
-    setup [:create_admin, :authenticate_admin]
-
-    test "renders form for editing chosen admin", %{conn: conn, admin: admin} do
-      conn = get(conn, Routes.admin_path(conn, :edit, admin))
-      assert html_response(conn, 200) =~ "Editar Admin"
+      assert json_response(conn, 422)["errors"] != %{}
```

Comment:
you could test if it is rendering the changeset errors

Reply:
garantir aqui no teste de controller que o errors não vai estar vazio não seria o suficiente? sendo que eu estou testando o changeset no teste de model? é uma boa pratica fazer um double check aqui tb?

Reply:
tudo bem, eu n sei qual é a boa prática, talvez gerar um scaffold de controller para ver como ele faz, nunca fiz

Reply:
Esse código foi baseado no scaffold

Reply:
🤝